src/TextInput.js

```diff
@@ -1,167 +1,161 @@
-import React, { Component } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class TextInput extends Component {
-  constructor(props) {
-    super(props);
+const TextInput = props => {
+  const [id, setId] = useState(props.id || idgen());
 
-    this.id = props.id || idgen();
+  const [idSetable, setIdSetable] = useState(true);
 
-    if (props.password) {
-      this.id = `password${idgen()}`;
-    }
+  if (props.password && idSetable) {
+    setId(`password${idgen()}`);
+    setIdSetable(false);
+  }
 
-    if (props.email) {
-      this.id = `email${idgen()}`;
-    }
+  if (props.email && idSetable) {
+    setId(`email${idgen()}`);
+    setIdSetable(false);
   }
 
-  componentDidMount() {
+  const inputRef = useRef(null);
+
+  // eslint-disable-next-line react/prop-types
+  const dataLength = props['data-length'];
+
+  useEffect(() => {
     if (typeof M !== undefined) {
-      // eslint-disable-next-line react/prop-types
-      this.props['data-length'] && M.CharacterCounter.init(this.inputRef);
+      dataLength && M.CharacterCounter.init(inputRef);
     }
-  }
+  }, [dataLength]);
 
-  componentDidUpdate(prevProps) {
-    const { value } = this.props;
-
-    if (value !== prevProps.value && typeof M !== 'undefined') {
+  useEffect(() => {
+    if (typeof M !== 'undefined') {
       M.updateTextFields();
     }
```

Comment:
we've been removing this check on other components, since with this check it will silently fail. Best if just `M.updateTextFields()`